junit-platform-launcher/src/main/java/org/junit/platform/launcher/listeners/TestExecutionSummary.java

```diff
@@ -127,7 +127,20 @@
 	 *
 	 * @see #printTo(PrintWriter)
 	 */
-	void printFailuresTo(PrintWriter writer);
+	default void printFailuresTo(PrintWriter writer) {
+		printFailuresTo(writer, 10);
+	}
+
+	/**
+	 * Print failed containers and tests, including sources and exception
+	 * messages, to the supplied {@link PrintWriter}.
+	 * The maximum number of lines to print for the exception stack traces
+	 * (if any) can be specified with the given {@code maxStackTraceLines}
+	 * parameter.
+	 *
+	 * @see #printTo(PrintWriter)
+	 */
+	void printFailuresTo(PrintWriter writer, int maxStackTraceLines);
```

Comment:
Please add `@param` entries for both parameters, including nullability information analogous to other places in the code base.

Reply:
The new method must be a `default` method, not the other way around.

Otherwise this will be a breaking change for anyone that may have implemented this interface (even if unlikely). So please make the appropriate changes to maintain backward compatibility.

Reply:
The new `printFailuresTo(PrintWriter, int)` method needs `@since` and `@API` declarations.

Reply:
Technically, the new `printFailuresTo(PrintWriter, int)` method must be a `default` method. So please go ahead and make it a `default` method that delegates to `printFailuresTo(PrintWriter)`.

Behaviorally, that seems a bit strange, since it could reasonably only delegate to `printFailuresTo(PrintWriter)` -- thereby ignoring the custom max stack trace lines -- or throw an `UnsupportedOperationException`.

I would personally go with the former and document that that is the case due to backward compatibility constraints.

@junit-team/junit-lambda, WDYT?